libyul/AsmParser.cpp

```diff
@@ -192,13 +196,67 @@ std::optional<std::pair<std::string_view, SourceLocation>> Parser::parseSrcComme
 	langutil::SourceLocation const& _commentLocation
 )
 {
-	static std::regex const argsRegex = std::regex(
-		R"~~(^(-1|\d+):(-1|\d+):(-1|\d+)(?:\s+|$))~~"  // index and location, e.g.: 1:234:-1
-		R"~~(("(?:[^"\\]|\\.)*"?)?)~~",                // optional code snippet, e.g.: "string memory s = \"abc\";..."
-		std::regex_constants::ECMAScript | std::regex_constants::optimize
-	);
-	std::match_results<std::string_view::const_iterator> match;
-	if (!regex_search(_arguments.cbegin(), _arguments.cend(), match, argsRegex))
+	std::string const s_args (_arguments);
+	CharStream argumentStream (s_args, "");
+	Scanner scanner (argumentStream);
+
+	int sourceIndex { -1 };
```

Comment:
This is more a nitpick, but I think overall in the code we usually don't have spaces before/after "constructor braces".
```suggestion
	int sourceIndex {-1};
```

Reply:
Yeah, this style makes things look so confusing. I see code blocks everywhere :P
```suggestion
	int sourceIndex = -1;
```

Reply:
haha ok ok, i see you're not fond of this style, i'll change it